File: java/05_029.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
import java.math.*;

public class Main {
    public static void main(String[] args) {
        // Main method code goes here
    }
    
    public class DirectedGraph {
        private Map<Object, List<Object>> graph;
        
        public DirectedGraph() {
            this.graph = new HashMap<>();
        }
        
        public void addRelationship(Object fromNode, Object toNode) {
            if (!this.graph.containsKey(fromNode)) {
                this.graph.put(fromNode, new ArrayList<>());
            }
            this.graph.get(fromNode).add(toNode);
        }
    }
}

File: data structures and algorithms/cs/05_029.cs
using System;
using System.Collections.Generic;

public class MainClass
{
   public static void Main(string[] args)
   {
      // Main method code goes here
   }

   public class DirectedGraph
   {
      private Dictionary<object, List<object>> graph;

      public DirectedGraph()
      {
         this.graph = new Dictionary<object, List<object>>();
      }

      public void AddRelationship(object fromNode, object toNode)
      {
         if (!this.graph.ContainsKey(fromNode))
         {
            this.graph[fromNode] = new List<object>();
         }
         this.graph[fromNode].Add(toNode);
      }
   }
}
